### ISSUE TITLE
Could not determine if the GCE Persistent Disk is attached in read-only or read-write mode

### ISSUE THREAD

**Issue by jas1989** [terraform-google-modules/terraform-google-container-vm#117]
### TL;DR

The container inside the VM doesn't start because it's unable to determine whether the mounted disk is in read-only or read-write mode

### Expected behavior

The expected behavior is as follows:

creation of the permanent disk
creation of the virtual machine
creation of the attached_disk resource
starting the container on the virtual machine

On the Google Console, you can see all the created resources, and the permanent disk indicates the mode (read/write).

### Observed behavior

All resources start, but the service responsible for launching the container returns the following error: `Error: Failed to start container: Volume dev-cronicle-data-disk: Could not determine if the GCE Persistent Disk dev-cronicle-data-disk is attached read-only or read-write.`

### Terraform Configuration

```hcl
resource "google_compute_disk" "pd" {
  ...
}

module "gce-container" {
  # https://github.com/terraform-google-modules/terraform-google-container-vm
  source = "terraform-google-modules/container-vm/google"
  version = "3.1.0"

  container = {
    image = "${var.dev-cronicle.image_cloud}:${var.dev-cronicle.version}"
    custom_command = [
      "./scripts/start-worker.sh"
    ]
    env = local.env_variables

    volumeMounts = [
      {
        mountPath = "/cache"
        name      = "tempfs-0"
        readOnly  = false
      },
      {
        mountPath = "/persistent-data"
        name      = "${google_compute_disk.pd.name}"
        readOnly  = false
      },
    ]
  }

  volumes = [
    {
      name = "tempfs-0"

      emptyDir = {
        medium = "Memory"
      }
    },
    {
      name = "${google_compute_disk.pd.name}"

      gcePersistentDisk = {
        pdName = "${google_compute_disk.pd.name}"
        fsType = "ext4"
      }
    },
  ]

  restart_policy = "Always"
}

resource "google_compute_attached_disk" "attached_disk" {
  disk        = google_compute_disk.pd.name
  instance    = google_compute_instance.vm.name
  device_name = "additional-disk" # Nome del disco nella macchina virtuale
  mode        = "READ_WRITE"
  zone        = var.zone_spec

  depends_on = [ 
    google_compute_disk.pd,
    google_compute_instance.vm
  ]
}

resource "google_compute_instance" "vm" {
  name = "dev-cronicle"
  machine_type = "e2-medium"
  zone    = "${var.zone_spec}"

  description = "metadata_for_recreation_compute_engine_version SHA512 hash (base64): ${base64sha512(local.metadata_for_recreation_compute_engine_version)}\nmetadata_for_recreation_compute_engine_variables SHA512 hash (base64): ${base64sha512(local.metadata_for_recreation_comupte_engine_variables)}"
  
  # If true, allows Terraform to stop the instance to update its properties.
  allow_stopping_for_update = true

  boot_disk {
    initialize_params {
      image = module.gce-container.source_image
    }
  }

  lifecycle {
    ignore_changes = [attached_disk]
  }

  network_interface {
    subnetwork_project = var.project_name
    subnetwork         = "default"
    access_config {
      nat_ip = google_compute_address.default.address
    }
  }

  metadata = {
    gce-container-declaration = module.gce-container.metadata_value
    metadata_for_recreation_compute_engine_version = local.metadata_for_recreation_compute_engine_version
    metadata_for_recreation_comupte_engine_variables = local.metadata_for_recreation_comupte_engine_variables
  }

  tags = ["dev-cronicle-container", "dev-cronicle-container-disk"]

  service_account {
    email = ""
    scopes = [
      "https://www.googleapis.com/auth/cloud-platform",
    ]
  }

  depends_on = [ 
    google_compute_disk.pd
  ]
}
```

### Terraform Version

```sh
Terraform v1.5.7
on linux_amd64

provider registry.terraform.io/hashicorp/google v4.57.0
provider registry.terraform.io/hashicorp/google-beta v5.0.0
```

**Comment by moritzschmitz-oviva** [comment]
@jas1989 Hey man, had the same problem today.

I got it to work by setting the device name to the same name as the pdDisk name.
